File: app/src/main/java/br/com/alura/ceep/ui/coffemachine/domain/CoffeMachineData.java
package br.com.alura.ceep.ui.coffemachine.domain;

import android.os.Parcel;
import android.os.Parcelable;

import java.io.Serializable;

@org.parceler.Parcel

public class CoffeMachineData implements Serializable {
    private String CoffeType;
    private String CoffeIntensity;
    private String CoffeIntensityTitule;
    private String CoffeDescription;
    private String CoffeSize;
    private String CoffeSizeTitule;

    public CoffeMachineData (String coffeType, String coffeIntensity, String coffeIntensityTitule,
                             String coffeDescription, String coffeSizeTitule, String coffeSize, Integer coffeImage) {
        CoffeType = coffeType;
        CoffeIntensity = coffeIntensity;
        CoffeIntensityTitule = coffeIntensityTitule;
        CoffeDescription = coffeDescription;
        CoffeSizeTitule = coffeSizeTitule;
        CoffeSize = coffeSize;
        CoffeImage = coffeImage;
    }

    protected CoffeMachineData (Parcel in) {
        CoffeType = in.readString ();
        CoffeIntensity = in.readString ();
        CoffeIntensityTitule = in.readString ();
        CoffeDescription = in.readString ();
        CoffeSize = in.readString ();
        CoffeSizeTitule = in.readString ();
        if (in.readByte () == 0) {
            CoffeImage = null;
        } else {
            CoffeImage = in.readInt ();
        }
    }

    public int describeContents () {
        return 0;
    }

    public static final Parcelable.Creator<CoffeMachineData> CREATOR = new Parcelable.Creator<CoffeMachineData> () {
        @Override
        public CoffeMachineData createFromParcel (Parcel in) {
            return new CoffeMachineData (in);
        }

        @Override
        public CoffeMachineData[] newArray (int size) {
            return new CoffeMachineData[size];
        }
    };

    public String getCoffeType () {
        return CoffeType;
    }

    public void setCoffeType (String coffeType) {
        CoffeType = coffeType;
    }

    public String getCoffeIntensity () {
        return CoffeIntensity;
    }

    public void setCoffeIntensity (String coffeIntensity) {
        CoffeIntensity = coffeIntensity;
    }

    public String getCoffeIntensityTitule () {
        return CoffeIntensityTitule;
    }

    public void setCoffeIntensityTitule (String coffeIntensityTitule) {
        CoffeIntensityTitule = coffeIntensityTitule;
    }

    public String getCoffeDescription () {
        return CoffeDescription;
    }

    public void setCoffeDescription (String coffeDescription) {
        CoffeDescription = coffeDescription;
    }

    public String getCoffeSizeTitule () {
        return CoffeSizeTitule;
    }

    public void setCoffeSizeTitule (String coffeSizeTitule) {
        CoffeSizeTitule = coffeSizeTitule;
    }

    public String getCoffeSize () {
        return CoffeSize;
    }

    public void setCoffeSize (String coffeSize) {
        CoffeSize = coffeSize;
    }

    public Integer getCoffeImage () {
        return CoffeImage;
    }

    public void setCoffeImage (Integer coffeImage) {
        CoffeImage = coffeImage;
    }

    private Integer CoffeImage;
}

File: app/src/main/java/br/com/alura/ceep/ui/coffemachine/domain/CoffeMachineData.kt
package br.com.alura.ceep.ui.coffemachine.domain

import android.widget.ImageView
import java.io.Serializable

class CoffeMachineData : Serializable {
    var Coffetype: String?
    var CoffeIntensity: String?
    var CoffeDescription: String?
    var CoffeSize: String?
    var CoffeImage: Int

    constructor(
        coffeType: String?, coffeIntensity: String?,
        coffeDescription: String?, coffeSize: String?, coffeImage: Int
    ) {
        Coffetype = coffeType
        CoffeIntensity = coffeIntensity
        CoffeDescription = coffeDescription
        CoffeSize = coffeSize
        CoffeImage = coffeImage
    }

//    protected constructor(`in`: Parcel) {
//        Coffetype = `in`.readString()
//        CoffeIntensity = `in`.readString()
//        CoffeIntensityTitule = `in`.readString()
//        CoffeDescription = `in`.readString()
//        CoffeSize = `in`.readString()
//        CoffeSizeTitule = `in`.readString()
//        if (`in`.readByte().toString()) {
//            CoffeImage = null
//        } else {
//            CoffeImage = `in`.readInt()
//        }
//    }

//    fun describeContents(): Int {
//        return 0
//    }


    //    companion object {
//        val CREATOR: Parcelable.Creator<CoffeMachineData> =
//            object : Parcelable.Creator<CoffeMachineData?> {
//                override fun createFromParcel(`in`: Parcel): CoffeMachineData? {
//                    return CoffeMachineData(`in`)
//                }
//
//                override fun newArray(size: Int): Array<CoffeMachineData?> {
//                    return arrayOfNulls(size)
//                }
//            }
//    }

//    open fun getCoffeType(): String? {
//        return coffeType
//    }
//
//    fun setCoffeType(coffeType: String) {
//        CoffeType = coffeType
//    }
//
//    fun getCoffeIntensity(): String? {
//        return coffeIntensity
//    }
//
//    fun setCoffeIntensity(coffeIntensity: String) {
//        CoffeIntensity = coffeIntensity
//    }
//
//    fun getCoffeIntensityTitule(): String? {
//        return CoffeIntensityTitule
//    }
//
//    fun setCoffeIntensityTitule(coffeIntensityTitule: String) {
//        CoffeIntensityTitule = coffeIntensityTitule
//    }
//
//    fun getCoffeDescription(): String? {
//        return CoffeDescription
//    }
//
//    fun setCoffeDescription(coffeDescription: String) {
//        CoffeDescription = coffeDescription
//    }
//
//    fun getCoffeSizeTitule(): String? {
//        return CoffeSizeTitule
//    }
//
//    fun setCoffeSizeTitule(coffeSizeTitule: String) {
//        CoffeSizeTitule = coffeSizeTitule
//    }
//
//    fun getCoffeSize(): String? {
//        return CoffeSize
//    }
//
//    fun setCoffeSize(coffeSize: String) {
//        CoffeSize = coffeSize
//    }
//
//    fun getCoffeImage(): Int? {
//        return CoffeImage
//    }
//
//    fun setCoffeImage(coffeImage: Int?) {
//        CoffeImage = coffeImage
//    }
//
//    private var CoffeImage: Int? = null

}
